Redis over TLS incompatible due to hardcoded tcp:// in entrypoint

### Context
Currently it is possible to use **Redis over TLS** by appending `tls://` to the `REDIS_FQDN` environment variable, which even if not present in the `template.env` file, is available in the code:
https://github.com/MISP/misp-docker/blob/67d4a0ea326155189c7549b451a1e55a213e2ab4/core/files/configure_misp.sh#L8
Therefore, by setting up this variable in the `.env` file, it is possible to communicate to an external Redis host, which would be via TLS if it's set up as `REDIS_FQDN: "tls://redis.example.com"`
### Error
Unfortunately, the `entrypoint_fpm.sh` file hardcodes the `tcp://` prefix in the FQDN:
https://github.com/MISP/misp-docker/blob/67d4a0ea326155189c7549b451a1e55a213e2ab4/core/files/entrypoint_fpm.sh#L22
Which makes PHP fail to connect to Redis, since the endpoint now looks like `tcp://tls://redis.example.com`.

If I'm having the same issue, it is related to lxml python setup for cybox:

`
Searching for lxml>=2.2.3
Reading https://pypi.python.org/simple/lxml/
Best match: lxml 4.4.1
Downloading https://files.pythonhosted.org/packages/c4/43/3f1e7d742e2a7925be180b6af5e0f67d38de2f37560365ac1a0b9a04c015/lxml-4.4.1.tar.gz#sha256=c81cb40bff373ab7a7446d6bbca0190bccc5be3448b47b51d729e37799bb5692
Processing lxml-4.4.1.tar.gz
Writing /tmp/easy_install-9a7Zl7/lxml-4.4.1/setup.cfg
Running lxml-4.4.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-9a7Zl7/lxml-4.4.1/egg-dist-tmp-jazmDM
/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
  warnings.warn(msg)
In file included from src/lxml/etree.c:692:0:
src/lxml/includes/etree_defs.h:14:31: fatal error: libxml/xmlversion.h: No such file or directory
compilation terminated.
/tmp/easy_install-9a7Zl7/lxml-4.4.1/temp/xmlXPathInitgop7jj.c:2:1: warning: return type defaults to ‘int’ [-Wimplicit-int]
 main (int argc, char **argv) {
 ^
Building lxml version 4.4.1.
Building without Cython.
Using build configuration of libxslt 1.1.28
ERROR: /usr/bin/xslt-config: 1: /usr/bin/xslt-config: xml2-config: not found

** make sure the development packages of libxml2 and libxslt are installed **

Compile failed: command 'x86_64-linux-gnu-gcc' failed with exit status 1
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
ERROR: Service 'web' failed to build: The command '/bin/sh -c python setup.py install' returned a non-zero code: 1
`

Resolved it by adding:
python-lxml python3-lxml to the apt-get setups, will open a PR.

--- a/core/files/entrypoint_fpm.sh
+++ b/core/files/entrypoint_fpm.sh
@@ -19,7 +19,7 @@ change_php_vars() {
         sed -i "s/upload_max_filesize = .*/upload_max_filesize = 50M/" "$FILE"
         sed -i "s/post_max_size = .*/post_max_size = 50M/" "$FILE"
         sed -i "s/session.save_handler = .*/session.save_handler = redis/" "$FILE"
-        sed -i "s|.*session.save_path = .*|session.save_path = 'tcp://${REDIS_FQDN}:6379'|" "$FILE"
+        sed -i "s|.*session.save_path =.*|session.save_path = '$(echo $REDIS_FQDN | grep -E '^\w+://' || echo tcp://$REDIS_FQDN):6379'|" "$FILE"
     done
 }
 

--- a/core/files/entrypoint_fpm.sh
+++ b/core/files/entrypoint_fpm.sh
@@ -19,7 +19,7 @@ change_php_vars() {
         sed -i "s/upload_max_filesize = .*/upload_max_filesize = 50M/" "$FILE"
         sed -i "s/post_max_size = .*/post_max_size = 50M/" "$FILE"
         sed -i "s/session.save_handler = .*/session.save_handler = redis/" "$FILE"
-        sed -i "s|.*session.save_path =.*|session.save_path = '$(echo $REDIS_FQDN | grep -E '^\w+://' || echo tcp://$REDIS_FQDN):6379'|" "$FILE"
+        sed -i "s|.*session.save_path = .*|session.save_path = '$(echo $REDIS_FQDN | grep -E '^\w+://' || echo tcp://$REDIS_FQDN):6379'|" "$FILE"
     done
 }
 

